### PR TITLE
check devices for all input tensors in index_put

### PR DESCRIPTION
**Pull request by ngimel**
Fix for #30960

### CODE REVIEW

File: aten/src/ATen/native/Indexing.cpp [hunk 1]
```diff
@@ -251,6 +251,10 @@ Tensor & _index_put_impl_(Tensor & self, TensorList indices, const Tensor & valu
     AT_INDEX_ERROR("too many indices for tensor of dimension ", self.dim(), " (got ", indices.size(), ")");
   }
   if (accumulate && self.device().type() == kCUDA) {
+      if (value.device() != self.device()) {
+        AT_ERROR("expected device ", self.device(), " but got device ",
```
Comment (mruberry) [inline aten/src/ATen/native/Indexing.cpp]: Isn't AT_ERROR deprecated in favor of TORCH_CHECK(false, ...)?

**Review by mruberry — APPROVED** [review]
Looks great!

File: aten/src/ATen/native/Indexing.cpp (hunk 1 above)
Comment (ngimel) [reply aten/src/ATen/native/Indexing.cpp]: You are right. There are 336 instances of AT_ERROR in ATen though.

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 0e8dc12:
* **1/1** broken upstream at merge base 9ca61ae (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=9ca61aec0f5569685257ec7497f739e17f7a3552&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=0e8dc12cddba001865bf981f5d78d9669b805c31).

### 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [caffe2_onnx_py3_6_clang7_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=75286181)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Pull request closed without merging**